corehq/messaging/scheduling/static/scheduling/js/conditional_alert_main.js

```diff
@@ -34,5 +34,34 @@ hqDefine("scheduling/js/conditional_alert_main", [
         // setup tab
         basicInformation.setRuleTabVisibility();
         $("#conditional-alert-basic-info-panel").koApplyBindings(basicInformation);
+
+        $("#conditional-alert-save-btn").click(function () {
+            var newRuleName = $("#id_conditional-alert-name").val();
+            if (!name) {
+                $("#conditional-alert-form").submit();
+            } else if (name !== newRuleName) {
```

Comment:
It looks like this will show the popup only if the user changed the name. So if the user left the name alone but changed the criteria or the scheduling info, that will kick off a rule run but the user won't be warned.

My read of the ticket is this popup should always show on save. If we later make updates so that rules don't run if only the name was changed, we'd then come back to this code and not show the popup in that scenario.

Reply:
yes, you are right, I am getting list of all the changed field using method Form.change_data without doing any configuration changes from UI side and stored it into a changed_schedule_field list. next time when request comes I am only considering fields which does not belongs to this list. `stop_type` also changes so it's not reliable.